test/SparqlExpressionTest.cpp

```diff
@@ -1091,15 +1091,16 @@ TEST(SparqlExpression, testToNumericExpression) {
   auto checkGetInt = testUnaryExpression<&makeConvertToIntExpression>;
   auto checkGetDouble = testUnaryExpression<&makeConvertToDoubleExpression>;
 
-  checkGetInt(idOrLitOrStringVec({U, "  -1275", "5.97", "-78.97", "-5BoB6",
-                                  "FreBurg1", "", " .", " 42\n", " 0.01 ", "",
-                                  "@", "@?+1", "1", G}),
-              Ids{U, I(-1275), U, U, U, U, U, U, I(42), U, U, U, U, I(1), U});
-  checkGetDouble(
-      idOrLitOrStringVec({U, "-122.2", "19,96", " 128789334.345 ", "-0.f",
-                          "  0.007 ", " -14.75 ", "Q", "@!+?", "1", G}),
-      Ids{U, D(-122.2), U, D(128789334.345), U, D(0.007), D(-14.75), U, U,
-          D(1.00), U});
+  checkGetInt(
+      idOrLitOrStringVec({U, "  -1275", "5.97", "-78.97", "-5BoB6", "FreBurg1",
+                          "", " .", " 42\n", " 0.01 ", "", "@", "@?+1", "1", G,
+                          "+42"}),
+      Ids{U, I(-1275), U, U, U, U, U, U, I(42), U, U, U, U, I(1), U, U});
+  checkGetDouble(idOrLitOrStringVec({U, "-122.2", "19,96", " 128789334.345 ",
+                                     "-0.f", "  0.007 ", " -14.75 ", "Q",
+                                     "@!+?", "1", G, "+42.0"}),
+                 Ids{U, D(-122.2), U, D(128789334.345), U, D(0.007), D(-14.75),
+                     U, U, D(1.00), U, D(42)});
```

Comment:
The tests for the `1E43` are missing (either here if we merge the casts, or in the (completely missing `checkGetDecimal` test.)
After our discussion, this test should include the case with scientific notation, and there should be a test for the `xsd:decimal` casting function, that it doesn't support the scientific notation.